Firestore/core/test/unit/bundle/bundle_serializer_test.cc

```diff
@@ -59,31 +73,39 @@ TEST(BundleSerializerTest, DecodesBundleMetadata) {
 }
 
 TEST(BundleSerializerTest, DecodesInvalidBundleMetadataReportsError) {
-  auto proto_metadata = ProtoBundleMetadata{};
-  *proto_metadata.mutable_id() = "bundle-1";
-  proto_metadata.mutable_create_time()->set_seconds(2);
-  proto_metadata.mutable_create_time()->set_nanos(3);
-  proto_metadata.set_version(1);
-  proto_metadata.set_total_bytes(123456789987654321L);
-  proto_metadata.set_total_documents(5);
+  auto proto_metadata = bundle_metadata();
+
   std::string json_string;
   MessageToJsonString(proto_metadata, &json_string);
   auto invalid = "123" + json_string;
 
   BundleSerializer serializer;
   ReadContext context;
-  serializer.DecodeBundleMetadata(&context, invalid);
+  serializer.DecodeBundleMetadata(context, invalid);
 
   EXPECT_FALSE(context.ok());
 
-  std::string to_replace("123456789987654321");
-  auto start = json_string.find(to_replace);
-  json_string.replace(start, to_replace.size(), "xxxyyyzzz");
+  // Replace total_bytes to a string unparseable to integer.
+  std::string json_copy =
+      replaced_copy(json_string, "123456789987654321", "xxxyyyzzz");
+  context = ReadContext{};
+  serializer.DecodeBundleMetadata(context, json_copy);
```

Comment:
Perhaps check that `context` is `ok` here? (applies after similar reassignments below)

Reply:
Done.